src/allow_std.rs

```diff
@@ -0,0 +1,94 @@
+use {AsyncRead, AsyncWrite};
+use bytes::{Buf, BufMut};
+use futures::{Async, Poll};
+use std::{fmt, io};
+
+/// A simple wrapper type which allows types which implement only
+/// implement `std::io::Read` or `std::io::Write`
+/// to be used in contexts which expect an `AsyncRead` or `AsyncWrite`.
+///
+/// If these types issue an error with the kind `io::ErrorKind::WouldBlock`,
+/// it is expected that they will notify the current task on readiness.
+/// Synchronous `std` types should not issue errors of this kind and
+/// are safe to use in this context. However, using these types with
+/// `AllowStdIo` will cause the event loop to block, so they should be used
+/// with care.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct AllowStdIo<T>(pub T);
```

Comment:
I think it would be better to make the inner representation of the struct private and add `AllowStdIo::new`. This allows the internals to be changed if needed.

Also, could you add`get_ref` / `get_mut` / `into_inner` implementations for this type?

Reply:
Done.